docs/source/conceptual-guides/endpoints.md

```diff
@@ -0,0 +1,180 @@
+  as an abstraction over the SQL database (code [here](https://github.com/mozilla-ai/lumigator/tree/31d4cd3ecdafba6292a71e125b31f23498ddf9da/lumigator/python/mzai/backend/backend/repositories)).
+  They make use of [record classes](https://github.com/mozilla-ai/lumigator/tree/31d4cd3ecdafba6292a71e125b31f23498ddf9da/lumigator/python/mzai/backend/backend/records) to refer to actual records in the database.
+
+![Lumigator Backend](../../assets/lumigator-backend.svg)
+
+```{admonition} Notation
+In the following, we will refer to paths inside Lumigator's repo relative to the
+`/lumigator/python/mzai` folder, e.g. the relative path `backend/api/routes/` (note the lack of a
+trailing slash) will map to the absolute path from the root of the repo
+`/lumigator/python/mzai/backend/backend/api/routes/`.
+```
+
+## Lumigator endpoints
+
+All the endpoints you can access in Lumigator's API are defined in
+[`backend/api/routes/`](https://github.com/mozilla-ai/lumigator/tree/b1ea63ba3e1aae5907e46ffbe9bfd809253c6053/lumigator/python/mzai/backend/backend/api/routes)
+and explicitly listed in
+[`backend/api/router.py`](https://github.com/mozilla-ai/lumigator/blob/b1ea63ba3e1aae5907e46ffbe9bfd809253c6053/lumigator/python/mzai/backend/backend/api/router.py),
+together with a metadata tag (defined [here](https://github.com/mozilla-ai/lumigator/blob/b1ea63ba3e1aae5907e46ffbe9bfd809253c6053/lumigator/python/mzai/backend/backend/api/tags.py))
+which is used to provide a short description of the route.
+
+Let us now walk through a few examples to understand how Lumigator's endpoints work.
+
+### The simplest endpoint: `/health`
+
+The [`/health`](https://github.com/mozilla-ai/lumigator/blob/b1ea63ba3e1aae5907e46ffbe9bfd809253c6053/lumigator/python/mzai/backend/backend/api/routes/health.py)
+route provides perhaps the simplest example as it allows you to get the current backend status which
+is a constant:
+
+```python
+@router.get("/")
+def get_health() -> HealthResponse:
+    return HealthResponse(deployment_type=settings.DEPLOYMENT_TYPE, status="OK")
+```
+
+Note that the returned type is a `HealthResponse`: this is a
+[pydantic model](https://docs.pydantic.dev/latest/api/base_model/) defining the schema of the
+returned data. The general rule is that all return values in our routes should match a predefined
+schema. Schemas are defined under the `schemas` directory (in particular, `HealthResponse` can be
+found [here](https://github.com/mozilla-ai/lumigator/blob/b1ea63ba3e1aae5907e46ffbe9bfd809253c6053/lumigator/python/mzai/schemas/schemas/extras.py#L16)),
+typically in files with the same name of the route, service, etc.
+
+Being this simple, all the code for `get_health()` directly appears in the route file. A
+`HealthResponse`, composed of a deployment type which is loaded from
+[`backend/settings.py`](https://github.com/mozilla-ai/lumigator/blob/b1ea63ba3e1aae5907e46ffbe9bfd809253c6053/lumigator/python/mzai/backend/backend/settings.py)
```

Comment:
Suggestion: link to line number of load https://github.com/mozilla-ai/lumigator/blob/b1ea63ba3e1aae5907e46ffbe9bfd809253c6053/lumigator/python/mzai/backend/backend/settings.py#L12